crates/secrets-client/src/lib.rs

```diff
@@ -0,0 +1,176 @@
+use async_nats::HeaderMap;
+use nkeys::XKey;
+use wasmcloud_secrets_types::{
+    Secret, SecretRequest, SecretResponse, RESPONSE_XKEY, WASMCLOUD_HOST_XKEY,
+};
+
+const DEFAULT_API_VERSION: &str = "v0";
+
+#[derive(Debug)]
+pub enum SecretClientError {
+    ConvertServerXkeyError,
+    ParseServerXkeyError,
+    RequestServerXkeyError,
+    InvalidXkeyError,
+    SealSecretRequestError,
+    SendSecretRequestError,
+    SerializeSecretRequestError,
+    ParseServerResponseXkeyError,
+    OpenSecretResponseError,
+    DeserializeSecretResponseError,
+    ServerError(String),
+    MissingSecretError,
+}
+
+impl std::fmt::Display for SecretClientError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                SecretClientError::ConvertServerXkeyError => "Convert Server Xkey Error",
+                SecretClientError::ParseServerXkeyError => "Parse Server Xkey Error",
+                SecretClientError::RequestServerXkeyError => "Request Server Xkey Error",
+                SecretClientError::InvalidXkeyError => "Invalid Xkey Error",
+                SecretClientError::SealSecretRequestError => "Seal SecretRequest Error",
+                SecretClientError::SendSecretRequestError => "Send SecretRequest Error",
+                SecretClientError::SerializeSecretRequestError => "Serialize SecretRequest Error",
+                SecretClientError::ParseServerResponseXkeyError => "Parse Server Response Error",
+                SecretClientError::OpenSecretResponseError => "Open SecretResponse Error",
+                SecretClientError::DeserializeSecretResponseError =>
+                    "Deserialize SecretResponse Error",
+                SecretClientError::ServerError(_) => "Server Error",
+                SecretClientError::MissingSecretError => "Missing Secret Errror",
+            }
+        )
+    }
+}
+
+impl std::error::Error for SecretClientError {}
+
+#[derive(Debug)]
+struct SecretsTopic(String);
+
+impl SecretsTopic {
+    pub(crate) fn new(prefix: &str, backend: &str, api_version: Option<&str>) -> Self {
+        let version = api_version.unwrap_or(DEFAULT_API_VERSION);
+        Self(format!("{}.{}.{}", prefix, version, backend))
+    }
+
+    pub fn get(&self) -> String {
+        format!("{}.{}", self.0, "get")
+    }
+
+    pub fn server_xkey(&self) -> String {
+        format!("{}.{}", self.0, "server_xkey")
+    }
+}
+
+#[derive(Debug)]
+pub struct Client {
+    client: async_nats::Client,
+    topic: SecretsTopic,
+    server_xkey: XKey,
+}
+
+impl Client {
+    pub async fn new(
```

Comment:
Since this is going to be used in the NATS impl and probably the community provided vault one, we probably should have this documented before we merge